src/nvim/os/env.c

```diff
@@ -423,43 +423,41 @@ static char *remove_tail(char *p, char *pend, char *name)
 /// @param[out] mustfree Ouput parameter for the caller to determine if they are
 ///           responsible for releasing memory. Must be initialized to false
 ///           by the caller.
-char_u *vim_getenv(char_u *name, bool *mustfree)
+char *vim_getenv(const char *name, bool *mustfree)
 {
-  char_u      *p;
-  char_u      *pend;
-  int vimruntime;
-
-
-  p = (char_u *)os_getenv((char *)name);
-  if (p != NULL && *p == NUL)       /* empty is the same as not set */
-    p = NULL;
-
-  if (p != NULL) {
-    return p;
+  const char *kos_env_path = os_getenv(name);
+  if (kos_env_path != NULL
+      && *kos_env_path == NUL) /* empty is the same as not set */
+    kos_env_path = NULL;
+
+  if (kos_env_path != NULL) {
+    *mustfree = true;
+    return xstrdup(kos_env_path);
   }
 
-  vimruntime = (STRCMP(name, "VIMRUNTIME") == 0);
-  if (!vimruntime && STRCMP(name, "VIM") != 0)
+  bool vimruntime = (strcmp(name, "VIMRUNTIME") == 0);
+  if (!vimruntime && strcmp(name, "VIM") != 0)
     return NULL;
 
   /*
    * When expanding $VIMRUNTIME fails, try using $VIM/vim<version> or $VIM.
    * Don't do this when default_vimruntime_dir is non-empty.
    */
+  char *vim_path = NULL;
   if (vimruntime
 #ifdef HAVE_PATHDEF
       && *default_vimruntime_dir == NUL
 #endif
       ) {
-    p = (char_u *)os_getenv("VIM");
-    if (p != NULL && *p == NUL)             /* empty is the same as not set */
-      p = NULL;
-    if (p != NULL) {
-      p = (char_u *)vim_version_dir((char *)p);
-      if (p != NULL)
-        *mustfree = true;
-      else
-        p = (char_u *)os_getenv("VIM");
+    kos_env_path = os_getenv("VIM");
+    if (kos_env_path != NULL
+        && *kos_env_path == NUL) /* empty is the same as not set */
```

Comment:
Crap - I knew the comment one.  Forgot about that.  I'll fix the braces...I knew I should've checked the style guide on that one, I had a nagging feeling.